File: 1l - Orientacao a objetos/java/comum/Joao.java
// class Joao herda de pessoa{
class Joao extends Pessoa{

	//construtor
	public Joao(){}
	public Joao(String cpf){
		this.cpf = cpf;
	}

	public static final String CONSTANTE = "valor fixo";

	// override de euSou - polimorfismo
	public void euSou(){
		System.out.println("eu sou joao");
	}

	// override de metodoSobresquever usando tb o retorno original
	public void metodoSobresquever(){
		System.out.println("sobresquevi");
		super.metodoSobresquever();
	}
}

File: 1l - Orientacao a objetos/ruby/joao.rb
require File.expand_path('../pessoa', __FILE__)

class Joao < Pessoa
	def initialize(cpf)
		@cpf = cpf
	end

	attr_accessor :cpf

	# override de euSou - polimorfismo
	def eu_sou
		puts("eu sou joao")
	end

	# override de metodoSobresquever usando tb o retorno original
	def metodo_sobresquever
		puts("sobresquevi")
		super
	end
end